### ISSUE TITLE
Feature Request: AutoConnect to Mobile Networks

### ISSUE THREAD

**Issue by GeneralTDog** [MarioNoll/WG-Auto-Connect#8]
Hi, I've noticed, that it successfully disconnects my VPN, when im in my designated network where it should disconnect.
It also sucessfully auto-enables the vpn tunnel when switching WiFi.

But i did notice it wont auto-enable when switching from my WLAN to a mobile network, so i would like to request this feature.

**Comment by MarioNoll** [comment]
Hey @GeneralTDog do I understand correctly that the enabling of the tunnel does not work for you when you disconnect from your designated WiFi? Does it work if you disable WiFi? Can you share some more details maybe? Which Android Version are you using?

**Comment by GeneralTDog** [comment]
Sure thing.

It does not establish a tunnel when:
- I'm disconnecting from designated WiFi to mobile
- I'm disabling designated WiFi and switching to mobile
However if I'm switching from designated WiFi to a different WiFi network it does establish a tunnel.

Android 14
LineageOS 21-20240426-NIGHTLY-kebab
lineage_kebab-userdebug 14 AP1A.240405.002.A1 6012ba0027

App via F-Droid
Version 1.1.0
de.marionoll.wgautoconnect

**Comment by Rouji** [comment]
Exactly the same thing here happening here, also on Lineage 21.

**Comment by MarioNoll** [comment]
@Rouji @GeneralTDog Can you please check if disabling "Pause app activity if unused" in the permission settings "Addtional Permissions"  for "control WireGuard tunnels" does help? After switching to a Pixel 8a I noticed the same behavior, for the moment that seems to have fixed it on my side.

**Comment by Rouji** [comment]
That does help a bit. It does enable the VPN now when switching to cellular, but not always. 
I'll keep an eye on it for a bit, maybe I'll notice a pattern

**Comment by MarioNoll** [comment]
Yeah I noticed the same flakiness, I did a little more digging and the request to the WireGuard App itself to activate the tunnel is send correctly, but somehow it didn't seem to get received. After disabling battery optimization for the WireGuard App though, it seems to work flawlessly now.

**Comment by Rouji** [comment]
I'll give that a try too

**Comment by Rouji** [comment]
Unfortunately didn't help.  
I can kind of confirm it seems to be more on the wireguard app's side though. It works perfectly for a short while after I've had the wireguard app open (or it was active for some other reason?) and then stops working again

**Comment by GeneralTDog** [comment]
Did some testing and disabled all power optimizations and checked all settings.
Unfortunately it indeed didn't work out for me aswell.
Also the Pause activity slider did sadly not fix it.

**Comment by Rouji** [comment]
> Unfortunately didn't help. I can kind of confirm it seems to be more on the wireguard app's side though. It works perfectly for a short while after I've had the wireguard app open (or it was active for some other reason?) and then stops working again

I'm gonna back-pedal on that, it does also work even if I've not touched the wireguard app in ages. It just (as far as I can tell) completely randomly sometimes fails to do its thing. 

Overall it does seem to work more often than not now though; the battery optimisation settings seem to have done *something*
